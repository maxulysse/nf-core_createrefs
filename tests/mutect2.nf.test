nextflow_pipeline {

    name "Test MUTECT2_PON"
    script "main.nf"
    tag "MUTECT2"

    test("Run MUTECT2 test") {

        when {
            params {
                outdir           = "$outputDir"
                tools            = 'mutect2'
                mutect2_pon_name = 'test'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_ceatepanelrefs_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
