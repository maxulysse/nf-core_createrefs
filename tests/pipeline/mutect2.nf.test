nextflow_pipeline {

    name "Test MUTECT2_PON"
    script "main.nf"
    tag "MUTECT2"

    test("Run MUTECT2 test") {

        when {
            params {
                outdir                       = "$outputDir"
                tools                        = 'mutect2'
                mutect2_pon_name             = 'test'
                validationSchemaIgnoreParams = 'genomes,baseDir,base-dir,outputDir,output-dir'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/multiqc/").exists() },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") }
            )
        }
    }
}
