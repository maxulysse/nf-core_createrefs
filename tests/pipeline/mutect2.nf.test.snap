{
    "software_versions": {
        "content": [
            "{GATK4_CREATESOMATICPANELOFNORMALS={gatk4=4.5.0.0}, GATK4_GENOMICSDBIMPORT={gatk4=4.5.0.0}, GATK4_MUTECT2={gatk4=4.5.0.0}, Workflow={nf-core/createpanelrefs=v1.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-04-03T17:06:08.093342"
    }
}